actionpack/test/dispatch/request/session_test.rb

```diff
@@ -103,6 +103,19 @@ def test_fetch
         end
       end
 
+      def test_with_indifferent_access
+        env = {}
+        s = Session.create(store, env, {})
+
+        s[:one] = { test: "deep" }
+        s[:two] = { "test" => "deep" }
+
+        s2 = Session.create(store, env, {})
```

Comment:
Very odd, you're 100% correct (I was using 2.1.2 but can't reproduce actually needing it). Thanks for your help!